src/Simulation/Common/QubitManager.cs

```diff
@@ -49,6 +61,91 @@ public class QubitManager
         // For qubits that are free, the array stores the index of the next free qubit (a value less than "Allocated" constant).
         //     The last free qubit in the list of free qubits stores the value "None".
 
+        private void UpdateQubitCapacity(long newQubitCapacity)
+        {
+            Debug.Assert(newQubitCapacity < MaxQubitCapacity);
+            NumQubits = newQubitCapacity;
+            None = NumQubits;
+            Allocated = NumQubits + 1;
+            Disabled = NumQubits + 2;
+            AllocatedForBorrowing = NumQubits + 3;
+            this.qubits = new long[NumQubits];
```

Comment:
Based on the way `UpdateQubitCapacity` is called during allocate, this array doubles in size for each allocation, but it's not clear to me that this extra space is needed.

Reply:
See the reply above.